File: sdkwork-ai-sdk-swift/Sources/API/Responses2Api.swift
import Foundation

public class Responses2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Cancel response
    public func cancelResponse(response_id: String) async throws -> CancelResponsePostResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/responses/\(response_id)/cancel"), body: nil)
        return response as? CancelResponsePostResponse
    }

    /// Create response
    public func createResponse(body: CreateResponseRequest) async throws -> CreateResponseResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/responses"), body: body)
        return response as? CreateResponseResponse
    }

    /// List response input items
    public func listResponseInputItems(response_id: String) async throws -> ListResponseInputItemsResponse? {
        let response = try await client.get(ApiPaths.aiPath("/v1/responses/\(response_id)/input_items"))
        return response as? ListResponseInputItemsResponse
    }

    /// Get response
    public func getResponse(response_id: String) async throws -> GetResponseResponse? {
        let response = try await client.get(ApiPaths.aiPath("/v1/responses/\(response_id)"))
        return response as? GetResponseResponse
    }

    /// Delete response
    public func deleteResponse(response_id: String) async throws -> DeleteResponseResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/v1/responses/\(response_id)"))
        return response as? DeleteResponseResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/Responses2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Responses2Api
    {
        private readonly HttpClient _client;

        public Responses2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Cancel response
        /// </summary>
        public async Task<CancelResponsePostResponse?> CancelResponseAsync(string response_id)
        {
            return await _client.PostAsync<CancelResponsePostResponse>(ApiPaths.AiPath($"/v1/responses/{response_id}/cancel"), null);
        }

        /// <summary>
        /// Create response
        /// </summary>
        public async Task<CreateResponseResponse?> CreateResponseAsync(CreateResponseRequest body)
        {
            return await _client.PostAsync<CreateResponseResponse>(ApiPaths.AiPath("/v1/responses"), body);
        }

        /// <summary>
        /// List response input items
        /// </summary>
        public async Task<ListResponseInputItemsResponse?> ListResponseInputItemsAsync(string response_id)
        {
            return await _client.GetAsync<ListResponseInputItemsResponse>(ApiPaths.AiPath($"/v1/responses/{response_id}/input_items"));
        }

        /// <summary>
        /// Get response
        /// </summary>
        public async Task<GetResponseResponse?> GetResponseAsync(string response_id)
        {
            return await _client.GetAsync<GetResponseResponse>(ApiPaths.AiPath($"/v1/responses/{response_id}"));
        }

        /// <summary>
        /// Delete response
        /// </summary>
        public async Task<DeleteResponseResponse?> DeleteResponseAsync(string response_id)
        {
            return await _client.DeleteAsync<DeleteResponseResponse>(ApiPaths.AiPath($"/v1/responses/{response_id}"));
        }
    }
}
